src/langcheck/metrics/en/pairwise_text_quality.py

```diff
@@ -0,0 +1,318 @@
+
+    def _function_call_prompt(long_assessment: str) -> str:
+        return f'''
+        The following is an assessment on whether Response A or Response B is
+        the better response to the user's query:
+        ************
+        [Assessment]: {long_assessment}
+        ************
+
+        Save the resulting assessment. The available assessments are:
+        `Response A`
+        `Response B`
+        `Tie`
+        '''
+
+    pairwise_comparison_assessment_to_score = {
+        'Response B': 1.0,
+        'Tie': 0.5,
+        'Response A': 0.0
+    }
+    oai_evaluator = OpenAIBasedEvaluator(
+        assessment_to_score_mapping=pairwise_comparison_assessment_to_score,
+        function_name='save_pairwise_comparison_assessment',
+        function_description=("Saves a pairwise comparison assessment."),
+        argument_name='pairwise_comparison',
+        argument_description='The pairwise comparison assessment',
+        client_type=model_type,
+        client=openai_client,
+        openai_args=openai_args)
+
+    score_list = []
+    explanation_list = []
+    prompt_fn, data_iter = _construct_pairwise_comparison_prompt_and_data(
+        generated_outputs_a, generated_outputs_b, prompts, sources_a, sources_b,
+        reference_outputs)
+    for data_instance in tqdm_wrapper(data_iter,
+                                      desc='Calculating scores',
+                                      total=len(prompts)):
+        score, explanation = oai_evaluator.get_score(prompt_fn(*data_instance),
+                                                     _function_call_prompt)
+        score_list.append(score)
+        explanation_list.append(explanation)
+
+    if enforce_consistency:
+        # Swap the generated outputs and calculate the scores again
+        swapped_score_list = []
+        swapped_explanation_list = []
+        prompt_fn, swapped_data_iter = _construct_pairwise_comparison_prompt_and_data(  # NOQA: E501
+            generated_outputs_b, generated_outputs_a, prompts, sources_b,
+            sources_a, reference_outputs)
+        for swapped_data_instance in tqdm_wrapper(
+                swapped_data_iter,
+                desc='Calculating scores with swapped inputs',
+                total=len(prompts)):
+            swapped_score, swapped_explanation = oai_evaluator.get_score(
+                prompt_fn(*swapped_data_instance), _function_call_prompt)
+            swapped_score_list.append(swapped_score)
+            swapped_explanation_list.append(swapped_explanation)
+
+        # Iterate through the scores and explanations to check for consistency.
+        # If a score is not consistent, set it to None, and merge the two
+        # explanations to show the inconsistency.
+        for i in range(len(score_list)):
+            if score_list[i] is None or swapped_score_list[i] is None:
+                # If either score is None, we cannot determine consistency, so
```

Comment:
From a practical perspective, it'll lead to more results not being `None`, which I think is a good thing.

From a "what's more principled" perspective, I agree that your way probably makes more sense. But the way that I've been thinking about this is `swapped_score_list` is used solely to validate the original results, so if there's no evidence of an inconsistency we should just keep the original results.

Another option that I considered was to use the one that is _not_ `None` as the result. However, this is a bit weird because the `explanation` would have `Model A` and `Model B` switched, which I think could be confusing.

(Note that `None` just means that something went wrong in the OpenAI pipeline, e.g. due to some connection error or Azure's filters rejecting an input.)

Reply:
Actually, maybe the better way to think about it is to only include the result if we can prove that the results are consistent, which we can't do if one of the results are None. I'll change this!

Reply:
Okay!

Reply:
Updated in https://github.com/citadel-ai/langcheck/pull/102/commits/1d1cbb94b52216c6556eaa497b8b3d0bc0a49453